internal/bminventory/inventory.go

```diff
@@ -3186,3 +3196,85 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
 	}
 	return buf.String(), nil
 }
+
+func (b *bareMetalInventory) RegisterOCPCluster(ctx context.Context) error {
+	log := logutil.FromContext(ctx, b.log)
+	id := strfmt.UUID(uuid.New().String())
+	url := installer.GetClusterURL{ClusterID: id}
+	clusterName := "ocp-assisted-service-cluster"
+
+	log.Infof("Register OCP cluster: %s with id %s", clusterName, id.String())
+
+	apiVIP, err := b.getApiVIPFromOCP(log)
+	if err != nil {
+		return err
+	}
+
+	openshiftVersion, err := b.getOpenshiftVersionFromOCP(log)
+	if err != nil {
+		return err
+	}
+
+	cluster := common.Cluster{Cluster: models.Cluster{
+		ID:               &id,
+		Href:             swag.String(url.String()),
+		Kind:             swag.String(models.ClusterKindAddHostsOCPCluster),
+		Name:             clusterName,
+		OpenshiftVersion: openshiftVersion,
+		UserName:         auth.UserNameFromContext(ctx),
+		OrgID:            auth.OrgIDFromContext(ctx),
+		UpdatedAt:        strfmt.DateTime{},
+		APIVipDNSName:    &apiVIP,
+	}}
+
+	err = validations.ValidateClusterNameFormat(clusterName)
+	if err != nil {
+		log.Errorf("Failed to validate cluster name: %s", clusterName)
+		return err
+	}
+
+	// Persist worker-ignition to s3 for cluster
+	err = b.createAndUploadNodeIgnition(ctx, &id, apiVIP)
+	if err != nil {
+		log.Errorf("Failed to create and upload worker ignition for cluster %s", id)
```

Comment:
```suggestion
		log.WithError(err).Errorf("Failed to create and upload worker ignition for cluster %s", id)
```
relevant to all the logs.

Reply:
fixed